fvm/fvm_bench_test.go

```diff
@@ -460,130 +523,288 @@ func BenchmarkRuntimeTransaction(b *testing.B) {
+		benchTransaction(b,
+			func(b *testing.B, context benchTransactionContext) string {
+				return templateTx(50, `let acct = AuthAccount(payer: signer)`)
+			},
+		)
 	})
 	b.Run("call empty contract function", func(b *testing.B) {
-		benchTransaction(b, templateTx(100, `TestContract.empty()`))
+		benchTransaction(b,
+			func(b *testing.B, context benchTransactionContext) string {
+				return templateTx(100, `TestContract.empty()`)
+			},
+		)
 	})
 	b.Run("emit event", func(b *testing.B) {
-		benchTransaction(b, templateTx(100, `TestContract.emit()`))
+		benchTransaction(b,
+			func(b *testing.B, context benchTransactionContext) string {
+				return templateTx(100, `TestContract.emit()`)
+			},
+		)
 	})
 	b.Run("borrow array from storage", func(b *testing.B) {
 		benchTransaction(
 			b,
-			templateTx(100, `
-				let strings = signer.borrow<&[String]>(from: /storage/test)!
-				var i = 0
-				while (i < strings.length) {
-				  log(strings[i])
-				  i = i +1
-				}
-			`),
+			func(b *testing.B, context benchTransactionContext) string {
+				return templateTx(100, `
+					let strings = signer.borrow<&[String]>(from: /storage/test)!
+					var i = 0
+					while (i < strings.length) {
+					  log(strings[i])
+					  i = i +1
+					}
+				`)
+			},
 		)
 	})
 	b.Run("copy array from storage", func(b *testing.B) {
 		benchTransaction(
 			b,
-			templateTx(100, `
-				let strings = signer.copy<[String]>(from: /storage/test)!
-				var i = 0
-				while (i < strings.length) {
-				  log(strings[i])
-				  i = i +1
+			func(b *testing.B, context benchTransactionContext) string {
+				return templateTx(100, `
+					let strings = signer.copy<[String]>(from: /storage/test)!
+					var i = 0
+					while (i < strings.length) {
+					  log(strings[i])
+					  i = i +1
+					}
+				`)
+			},
+		)
+	})
+
+	b.Run("evm", func(b *testing.B) {
+
+		benchTransaction(
+			b,
+			func(b *testing.B, context benchTransactionContext) string {
+				coinbaseBytes := context.EvmTestAccount.Address().Bytes()
+				transactionBody := fmt.Sprintf(`
+				                    let coinbaseBytesRaw = "%s".decodeHex()
+					let coinbaseBytes: [UInt8; 20] = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+					for j, v in coinbaseBytesRaw {
+						coinbaseBytes[j] = v
+					}
+					let coinbase = EVM.EVMAddress(bytes: coinbaseBytes)
+				`, hex.EncodeToString(coinbaseBytes))
+
+				num := int64(12)
+				gasLimit := uint64(100_000)
+
+				// add 10 EVM transactions to the Flow transaction body
+				for i := 0; i < 100; i++ {
+					txBytes := context.EvmTestAccount.PrepareSignAndEncodeTx(b,
+						context.EvmTestContract.DeployedAt.ToCommon(),
+						context.EvmTestContract.MakeCallData(b, "store", big.NewInt(num)),
+						big.NewInt(0),
+						gasLimit,
+						big.NewInt(0),
+					)
+					transactionBody += fmt.Sprintf(`
+						let txBytes%[1]d = "%[2]s".decodeHex()
+						EVM.run(tx: txBytes%[1]d, coinbase: coinbase)
+						`,
+						i,
+						hex.EncodeToString(txBytes),
+					)
```

Comment:
could these two be combined and just have a boolean that adds the run line

Reply:
good point!